ironfish-rust/src/keys/public_address.rs

```diff
@@ -110,10 +94,9 @@ impl PublicAddress {
 
     /// Retrieve the public address in byte form. It is comprised of the
     /// 11 byte diversifier followed by the 32 byte transmission key.
-    pub fn public_address(&self) -> [u8; 43] {
-        let mut result = [0; 43];
-        result[..11].copy_from_slice(&self.diversifier.0);
-        result[11..].copy_from_slice(
+    pub fn public_address(&self) -> [u8; 32] {
+        let mut result = [0; 32];
+        result[..32].copy_from_slice(
             &point_to_bytes(&self.transmission_key)
                 .expect("transmission key should be convertible to bytes"),
         );
```

Comment:
Given memory is expensive to allocate and `point_to_bytes` already allocates a 32-byte array, it is possible to just use the return value of point_to_bytes directly

```suggestion
        let result = point_to_bytes(&self.transmission_key)
                .expect("transmission key should be convertible to bytes")
```

Reply:
Also do we want this to have an error condition instead of a crash? Crashing across boundaries is normally undefined behavior.